src/main/kotlin/net/atos/zac/app/zaken/converter/historie/MapUtils.kt

```diff
@@ -2,7 +2,11 @@ package net.atos.zac.app.zaken.converter.historie
 
 import net.atos.client.zgw.shared.util.JsonbUtil
 
-fun Map<*, *>.diff(other: Map<*, *>): Map<Any?, Pair<Any?, Any?>> = other
+fun Any?.asMapWithKeyOfString(): Map<String, *>? = (this as? Map<*, *>)
```

Comment:
are there not too many ?'s in here? E.g. is the `as?` really needed?

Reply:
If we use a regular `as`, it would throw if it's not a map, so then we would have to carefully document that you should absolutely sure that the thing you're calling on is definitely a map.

Reply:
In general I think I would say: if you are not a map why would you be calling this function in the first place? It does not do anything in that case right? If feasible I think it is generally to filter this out before actually calling this function, but I realise that may not always be so easy.

Reply:
Yeah, maybe it's doing too much now? We could extend `Map<*,*>` in stead and use the function like this:
```kotlin
fun Map<*,*>.asMapWithKeyOfString(): Map<String, *> = 
    this.mapNotNull { (key, value) -> (key as? String)?.let { it to value } }
        .toMap()

val maybeMap = (myAny as? <*,*>)?.asMapWithKeyOfString()
```

Reply:
Looks a bit better yes. :-) Also fine if you merge the PR as is. We can always improve afterwards.